agent/engine/service_connect/manager_linux.go

```diff
@@ -64,24 +71,37 @@ func NewManager() Manager {
 		statusPathHostRoot:  defaultStatusPathHostRoot,
 		statusFileName:      defaultStatusFileName,
 		statusENV:           defaultStatusENV,
+
+		adminStatsRequest: defaultAdminStatsRequest,
+		adminDrainRequest: defaultAdminDrainRequest,
 	}
 }
 
-func (m *manager) initializeAgentContainer(task *apitask.Task, container *apicontainer.Container, hostConfig *dockercontainer.HostConfig) error {
+func (m *manager) augmentAgentContainer(task *apitask.Task, container *apicontainer.Container, hostConfig *dockercontainer.HostConfig) error {
 	if task.IsNetworkModeBridge() {
 		err := m.initServiceConnectContainerMapping(task, container, hostConfig)
 		if err != nil {
 			return err
 		}
 	}
-	return m.initServiceConnectDirectoryMounts(task.GetID(), container, hostConfig)
+	adminPath, err := m.initAgentDirectoryMounts(task.GetID(), container, hostConfig)
+	if err != nil {
+		return err
+	}
+	m.initAgentEnvironment(container)
+
+	// Setup runtime configuration
+	task.ServiceConnectConfig.RuntimeConfig.AdminSocketPath = adminPath
```

Comment:
I'm thinking that since we need to read these values from a different goroutine in the future (stats engine), we might need to protect these with the task mutex. To that end, maybe it's best to have setter//getter methods for these.